R/validate_login.R

```diff
@@ -0,0 +1,35 @@
+#' Check ETN Database Credentials
+#'
+#' @description
+#' Validates user credentials for the ETN database by attempting to establish
+#' a connection. Returns TRUE if the credentials are valid and FALSE otherwise.
+#'
+#' @param username character string containing the ETN database username
+#' @param password character string containing the ETN database password
+#'
+#' @return logical; TRUE if credentials are valid and connection successful,
+#'         FALSE if authentication fails
+#'
+#' @examples
+#' \dontrun{
+#' # Check if credentials are valid
+#' is_valid <- check_credentials("my_username", "my_password")
+#' }
+#' @export
+validate_login <- function(username, password) {
+  assertthat::assert_that(!missing(username),
```

Comment:
Does `assert_that` support classed errors? If so, I would provide a:

```
class = "etnservice_error_username_missing`
```

Same for missing password.

Reply:
I made an issue #70 to switch to error classes, this will be part of a larger effort for etn and etnservice.